R/create_service_use_lookup.R

```diff
@@ -0,0 +1,777 @@
+#' and the specialty is not AB
+#'
+#' @inheritParams calculate_geriatric_cost
+#' @param cij_pattype A vector of CIJ patient types
+#' @param cij_ipdc A vector of CIJ inpatient/daycase markers
+#'
+#' @return A vector of Acute Elective costs
+#' @family Demographic and Service Use Cohort functions
+calculate_acute_elective_cost <- function(recid, cij_pattype, cij_ipdc, spec, cost_total_net) {
+  acute_elective_cost <- dplyr::if_else(recid %in% c("01B") &
+    (cij_pattype %in% c("Elective") | cij_ipdc %in% c("D")) &
+    !(spec %in% c("AB")), cost_total_net, 0)
+  return(acute_elective_cost)
+}
+
+#' Calculate cost for Acute Emergency records
+#' @description A record is considered to have an Acute Emergency cost if
+#' the recid is 01B, the CIJ patient type is Non-Elective, and the specialty is not AB
+#'
+#' @inheritParams calculate_geriatric_cost
+#' @param cij_pattype A vector of CIJ patient types
+#'
+#' @return A vector of Acute Emergency costs
+#' @family Demographic and Service Use Cohort functions
+calculate_acute_emergency_cost <- function(recid, cij_pattype, spec, cost_total_net) {
+  acute_emergency_cost <- dplyr::if_else(recid %in% c("01B") & cij_pattype %in% c("Non-Elective") &
+    !(spec %in% c("AB")), cost_total_net, 0)
+  return(acute_emergency_cost)
+}
+
+#' Calculate cost for Outpatient records
+#' @description A record is considered to have an Outpatient cost if the recid is 00B. The
+#' Outpatient Cost is defined as (total net cost - geriatric cost), and the Total Outpatient Cost
+#' is the total net cost
+#'
+#' @param recid A vector of record IDs
+#' @param cost_total_net A vector of total net costs
+#' @param geriatric_cost A vector of geriatric care costs
+#'
+#' @return A list with outpatient costs in index 1 and total outpatient costs in index 2
+#' @family Demographic and Service Use Cohort functions
+calculate_outpatient_costs <- function(recid, cost_total_net, geriatric_cost) {
+  outpatient_cost <- dplyr::if_else(
+    recid %in% c("00B"), cost_total_net - geriatric_cost, 0
+  )
+  total_outpatient_cost <- dplyr::if_else(
+    recid %in% c("00B"), cost_total_net, 0
+  )
+  return(list(
+    outpatient_cost = outpatient_cost,
+    total_outpatient_cost = total_outpatient_cost
+  ))
+}
+
+#' Calculate cost for Home Care records
+#' @description Please note: this function is currently not in use.
+#' A record is considered to have a home care cost if the recid is in
+#' c("HC-", "HC + ", "INS", "RSP", "MLS", "DC", "CPL")
+#'
+#' @param recid A vector of record IDs
+#' @param cost_total_net A vector of total net costs
+#'
+#' @return A vector of home care costs
+#' @family Demographic and Service Use Cohort functions
+calculate_home_care_cost <- function(recid, cost_total_net) {
+  home_care_cost <- dplyr::if_else(
+    recid %in% c("HC-", "HC + ", "INS", "RSP", "MLS", "DC", "CPL"), cost_total_net, 0
+  )
+  return(home_care_cost)
+}
```

Comment:
This isn't properly implemented yet so should just return 0
```suggestion
calculate_home_care_cost <- function(recid, cost_total_net) {
  #home_care_cost <- dplyr::if_else(
  # recid %in% c("HC-", "HC + ", "INS", "RSP", "MLS", "DC", "CPL"), cost_total_net, 0
  #)
  
  home_care_cost <- 0
  return(home_care_cost)
}
```